libsnark/gadgetlib1/gadgets/verifiers/kzg10_verifier_gadget.hpp

```diff
@@ -0,0 +1,134 @@
+/** @file
+ *****************************************************************************
+ * @author     This file is part of libff, developed by Clearmatics Ltd
+ *             (originally developed by SCIPR Lab) and contributors
+ *             (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#ifndef LIBSNARK_GADGETLIB1_GADGETS_VERIFIERS_KZG10_VERIFIER_GADGET_HPP_
+#define LIBSNARK_GADGETLIB1_GADGETS_VERIFIERS_KZG10_VERIFIER_GADGET_HPP_
+
+#include "libsnark/gadgetlib1/gadgets/curves/weierstrass_g1_gadget.hpp"
+#include "libsnark/gadgetlib1/gadgets/curves/weierstrass_g2_gadget.hpp"
+#include "libsnark/gadgetlib1/gadgets/pairing/pairing_checks.hpp"
+#include "libsnark/gadgetlib1/protoboard.hpp"
+#include "libsnark/polynomial_commitments/kzg10.hpp"
+
+/// Reference:
+/// - [KZG10]:
+///   Title: "Constant-Size Commitments to Polynomials and Their Applications"
+///   eprint: https://www.iacr.org/archive/asiacrypt2010/6477178/6477178.pdf
+
+namespace libsnark
+{
+
+/// The SRS for the KZG10 scheme, as protoboard variables. Names match those
+/// used in kzg10.hpp.
+template<typename ppT> class kzg10_srs_variable
+{
+public:
+    using npp = other_curve<ppT>;
+
+    std::vector<G1_variable<ppT>> alpha_powers_g1;
+    G2_variable<ppT> alpha_g2;
+
+    kzg10_srs_variable(
+        protoboard<libff::Fr<ppT>> &pb,
+        const size_t max_degree,
+        const std::string &annotation_prefix);
+
+    void generate_r1cs_witness(const typename kzg10<npp>::srs &srs);
+};
+
+/// The polynomial for the KZG10 scheme, as protoboard variables. This is just
+/// a G1_variable (see kzg10.hpp).
+template<typename ppT> using kzg10_commitment_variable = G1_variable<ppT>;
+
+/// The witness for the evaluation of a polynomial, as protoboard variables.
+/// This is also a single G1_variable (see kzg10.hpp).
```

Comment:
You may want to drop the path to the file: https://github.com/clearmatics/libsnark/blob/develop/libsnark/polynomial_commitments/kzg10.hpp or at least part of it, like: `libsnark/polynomial_commitments/kzg10.hpp`

Reply:
To avoid repeating `libsnark/polynomial_commitments/kzg10.hpp` everywhere, you can add a line after the references above to mention that the native implementation is at `libsnark/polynomial_commitments/kzg10.{hpp, tcc}`

Reply:
Added a line after references pointing to the native impl.  Changed references to the file name to references to "the native implementation".